datafusion/catalog/src/session.rs

```diff
@@ -145,7 +145,7 @@ impl From<&dyn Session> for TaskContext {
             state.scalar_functions().clone(),
             state.aggregate_functions().clone(),
             state.window_functions().clone(),
-            state.runtime_env().clone(),
+            Arc::clone(state.runtime_env()),
```

Comment:
this shows what the lint disallows -- when it is an `Arc` it must use `Arc::clone` rather than `.clone()` so it is easier to see when reading the code that no copies are made

Reply:
I always wondered what the motivation of the lint is 😆

Reply:
That is a good point -- I'll add some comments explaining it